feature/selector.py

```diff
@@ -448,14 +577,21 @@ def _validate_args(seed, selection_method) -> NoReturn:
 
     def _validate_fit(self, data, labels):
 
+        check_true(data is not None, ValueError("Data cannot be none"))
+
         # VIF is a Statistical methods, hence BaseSupervised, but does not need labels
         if isinstance(self._imp, _Statistical) and self.selection_method.method == "variance_inflation":
             pass
         else:
             # Supervised implementors, except VIF, require labels
             if isinstance(self._imp, _BaseSupervisedSelector):
                 check_true(labels is not None, ValueError("Labels column cannot be none"))
-                check_true(isinstance(labels, pd.Series), ValueError("Labels should be a pandas series/column."))
+
+                # For text_based labels is a dataframe
+                if isinstance(self._imp, _TextBased):
+                    check_true(isinstance(labels, pd.DataFrame), ValueError("Labels should be a pandas dataframe."))
```

Comment:
you can also assert that the number of columns in "data" and "labels" should match each other.

Reply:
Added!

Reply:
I think I dropped a comment related in the Tests, that, we can check that invalid sizes between data and label is handled nicely

Reply:
Added!